superset/models.py

```diff
@@ -2178,6 +2206,9 @@ def get_having_filters(self, raw_filters):
                 filters = cond
         return filters
 
+sqla.event.listen(DruidDatasource, 'before_insert', set_perm)
```

Comment:
datasource_name / cluster rename will change the permission and will revoke the user access to it, just an observation.